pkg/tls/certificatemanagement/assets.go

```diff
@@ -0,0 +1,90 @@
+package certificatemanagement
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/dns"
+)
+
+const (
+	// PrometheusServerTLSSecretName is the TLS secret used when the authentication proxy container in tigera-prometheus
+	// is acting as a server and handles incoming requests.
+	PrometheusServerTLSSecretName = "tigera-prometheus-server-tls"
+)
+
+// TLSAsset should be instantiated at startup only, none of its members are exposed as they should be immutable.
+type TLSAsset struct {
+	secretName         string
+	serviceaccountName string
+	namespace          string
+	permittedDNSNames  []string
+	keyExtUsages       []x509.ExtKeyUsage
+}
+
+func (t TLSAsset) SecretName() string {
+	return t.secretName
+}
+
+func (t TLSAsset) DNSNames() []string {
+	return t.permittedDNSNames
+}
+
+func (t TLSAsset) ExtKeyUsages() []x509.ExtKeyUsage {
+	return t.keyExtUsages
+}
+
+// TLSAssetMap can look up all assets associated with a serviceaccount and determine if a CSR is requesting allowed
+// certificate contents.
+type TLSAssetMap map[string]TLSAsset
+
+var (
+	// TLSAssets Stores the secret name as key and asset as value.
+	TLSAssets           = make(TLSAssetMap)
+	PrometheusServerTLS TLSAsset
+)
+
+// ValidRequestor returns true if the TLS may be created by the service account by means of a CSR.
+func (t TLSAssetMap) ValidRequestor(csrName, requestor string) bool {
+	secretName := strings.Split(csrName, ":")[0] // The CSRName is formatted as "<secretName>:<podName>".
+	if asset, found := t[secretName]; found {
+		return fmt.Sprintf("system:serviceaccount:%s:%s", asset.namespace, asset.serviceaccountName) == requestor
+	}
+	return false
+}
+
+// ValidDNSNames returns true if the request only contains permitted dns names for given secretName.
+func (t TLSAssetMap) ValidDNSNames(csrName string, request *x509.CertificateRequest) bool {
+	secretName := strings.Split(csrName, ":")[0] // The CSRName is formatted as "<secretName>:<podName>".
+	if asset, found := t[secretName]; found {
+		// Create this map at runtime, rather than keeping it in memory as CSRs are issued sporadically.
+		permittedNames := make(map[string]bool, len(asset.permittedDNSNames))
+		for _, permitted := range asset.permittedDNSNames {
+			permittedNames[permitted] = true
+		}
+		for _, name := range request.DNSNames {
+			if !permittedNames[name] {
+				return false
+			}
+		}
+		if request.Subject.CommonName != "" && !permittedNames[request.Subject.CommonName] {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+// InstantiateTLSAssets Called when the operator initializes.
```

Comment:
The fact this is called when the operator starts is obvious from the code, so this comment doesn't add anything. Why is it called then? What does this function actually do and when should I modify it?